server/test/services/geo/esri/EsriClientTest.java

```diff
@@ -177,4 +182,83 @@ public void getAddressSuggestionsWithError() throws Exception {
     assertThat(group.getWellKnownId()).isEqualTo(0);
     assertThat(group.getOriginalAddress()).isEqualTo(address);
   }
+
+  @Test
+  public void verifyMappingAddressFromJsonAttributes_useRegionAbbrField()
+      throws JsonProcessingException {
+    String json =
+        "{\"attributes\": {"
+            + "  \"SubAddr\": \"line2-expected\","
+            + "  \"Address\": \"street-expected\","
+            + "  \"City\": \"city-expected\","
+            + "  \"Region\": null,"
+            + "  \"RegionAbbr\": \"WA\","
+            + "  \"Postal\": \"11111-expected\""
+            + "}}";
+
+    runMapAddressAttributesJson(
+        json, "street-expected", "line2-expected", "city-expected", "WA", "11111-expected");
+  }
+
+  @Test
+  public void verifyMappingAddressFromJsonAttributes_useRegionField()
+      throws JsonProcessingException {
+    String json =
+        "{\"attributes\": {"
+            + "  \"SubAddr\": \"line2-expected\","
+            + "  \"Address\": \"street-expected\","
+            + "  \"City\": \"city-expected\","
+            + "  \"Region\": \"WA\","
+            + "  \"RegionAbbr\": \"Washington\","
+            + "  \"Postal\": \"11111-expected\""
+            + "}}";
+
+    runMapAddressAttributesJson(
+        json, "street-expected", "line2-expected", "city-expected", "WA", "11111-expected");
+  }
+
+  @Test
+  public void verifyMappingAddressFromJsonAttributes_useLine2AsEnteredIfNull()
+      throws JsonProcessingException {
+    String json =
+        "{\"attributes\": {"
+            + "  \"SubAddr\": null,"
+            + "  \"Address\": \"street-expected\","
+            + "  \"City\": \"city-expected\","
+            + "  \"Region\": \"WA\","
+            + "  \"RegionAbbr\": \"Washington\","
+            + "  \"Postal\": \"11111-expected\""
+            + "}}";
+
+    runMapAddressAttributesJson(
+        json, "street-expected", "line2-user", "city-expected", "WA", "11111-expected");
+  }
+
+  private void runMapAddressAttributesJson(
```

Comment:
**suggestion:** I'm not sure what the best name is, but thoughts on calling this something like runAndVerifyMapAddressAttributesJson or something? I think with the name its unclear this function is doing the asserts as well.